llumnix/entrypoints/vllm/api_server.py

```diff
@@ -237,9 +241,15 @@ async def is_ready():
 
     logger.info("engine_args: {}".format(engine_args))
 
+    set_runtime_env(engine_manager_args)
+
     if args.launch_ray_cluster:
         # Launch the ray cluster for multi-node serving.
         launch_ray_cluster(args.ray_cluster_port)
+    else:
+        # Connect to a ray cluster.
+        head_node_ip = os.getenv('HEAD_NODE_IP')
+        ray.init(address=f"{head_node_ip}:{args.ray_cluster_port}", ignore_reinit_error=True, namespace="llumnix")
```

Comment:
We can move ray.init outside of launch_ray_cluster. And call ray.init in a separate connect_to_ray_cluster function no matter launch_ray_cluster is True or False.